### DISCUSSION TITLE
AutoComplete: Not compatible with DataTable rowStatePreserved=true

### DISCUSSION THREAD

**Discussion by fanste** [primefaces/community#1926] [PrimeFaces]
As we just encountered this in one of our applications:

- AutoComplete searchs for completion items via event (`#broadcast`) and caches the result in the component variable `suggestions`.
- The cached items are then rendered during the render response phase.
- The variable is cleared on `#saveState`

This works as long as da parent component does normal state saving of `EditableValueHolder` properties only. For `DataTable` this is the case with `rowStatePreserved=false`. With `rowStatePreserved=true` `AutoComplete#saveState` is invoked after broadcast because of `DataTable#setRowIndex` which then clears the variable and nothing is available to render.

Is this a known behavior? I don't think, that this is an easy fixable case (@tandraschko you are the one, who prepared the code for MyFaces view pooling, any idea on that?).
If it could be fixed, I can open an issue.

Currently I don't know, why we used `rowStatePreserved=true` in our case. Right now, we can't spot any miss-behaviour in the remaining app logic.

**Comment by melloware** [comment]
@fanste do you know a fix?  If so open an issue and PR?

**Reply by fanste** [reply]
No, not yet. As "#saveState" is somewhat stateless and has no information about such usecases (at least I think so).
